src/main/java/org/jabref/gui/util/BackgroundTask.java

```diff
@@ -27,14 +33,29 @@
  * @param <V> type of the return value of the task
  */
 public abstract class BackgroundTask<V> {
+
+    public static ImmutableMap<String, Node> iconMap = ImmutableMap.of(
```

Comment:
Why do you prefer a map here instead of adding a new property `icon` to the BackgroundTask class (so that `downloadTask.setIcon(IconTheme.JabRefIcons.DOWNLOAD)` works)?

Reply:
That's because the TaskProgressView works on a list of tasks. Tasks have no icon property, therefore one must provide a callback that gets an icon from a task.
If BackgroundTask were derived from Task, we could add a property to BackgroundTask. The callback could then check if we have a BackgroundTask at hand and if so, provide the icon stored in a property in BackgroundTask. But BackgroundTask is not derived from Task so this is not possible.
We need a key to map from a Task to an Icon.
I used the title of the task as a key property to map to the icon, so an immutable map seemed like the best way to go.

Reply:
Ok, I see. It's a bit unfortunate but your solution makes sense. I would propose to change the `Callback` below to a normal method (`public static Node getIcon(Task<?> task)`) and replace the `iconMap` by a normal switch statement.

Reply:
Change to method is easy, but I cannot use a switch, because I would need a constant expression. As I use the title of the task, and there is localization on that, it is not constant.
https://github.com/JabRef/jabref/pull/6443/commits/396411a68cf3a2583cbe367b716d6195f0773cf2